editor/src/components/navigator/left-pane.tsx

```diff
@@ -763,6 +768,259 @@ const GithubSpinner = () => {
   )
 }
 
+interface RepositoryRowProps extends RepositoryEntry {
+  importPermitted: boolean
+}
+
+const RepositoryRow = (props: RepositoryRowProps) => {
+  const colorTheme = useColorTheme()
+
+  const dispatch = useEditorState((store) => store.dispatch, 'RepositoryRow dispatch')
+
+  const projectID = useEditorState((store) => store.editor.id, 'RepositoryRow projectID')
+
+  const importRepository = React.useCallback(() => {
+    const parsedTargetRepository = parseGithubProjectString(props.fullName)
+    if (parsedTargetRepository == null || props.defaultBranch == null) {
+      dispatch(
+        [
+          showToast(
+            notice(
+              `Error when attempting to import a repository with repo: ${props.fullName}`,
+              'ERROR',
+            ),
+          ),
+        ],
+        'everyone',
+      )
+    } else {
+      void getBranchContent(
+        dispatch,
+        parsedTargetRepository,
+        forceNotNull('Should have a project ID.', projectID),
+        props.defaultBranch,
+      )
+    }
+  }, [dispatch, projectID, props.fullName, props.defaultBranch])
+
+  return (
+    <div
+      style={{
+        minHeight: 40,
+        display: 'flex',
+        flexDirection: 'row',
+        gap: 8,
+        alignItems: 'center',
+        paddingLeft: 8,
+        paddingRight: 8,
+        paddingBottom: 8,
+        paddingTop: 8,
+        borderBottom: '1px solid #ccc',
+      }}
+    >
+      <div
+        style={{
+          borderRadius: '50%',
+          width: 20,
+          height: 20,
+          border: '1px solid #ccc',
+          backgroundImage: `url(${props.avatarUrl})`,
+          backgroundPosition: 'center',
+          backgroundSize: 'cover',
+          backgroundRepeat: 'no-repeat',
+        }}
+      />
+      <div style={{ flexGrow: 1 }}>
+        <span style={{ fontWeight: 600, textOverflow: 'ellipsis' }}>
+          {props.name ?? props.fullName}
+        </span>{' '}
+        <br />
+        <span style={{ opacity: 0.5 }}>
+          {props.private ? 'private' : 'public'}
+          {props.updatedAt == null ? null : (
+            <>
+              {' · '}
+              <TimeAgo date={props.updatedAt} />
+            </>
+          )}
+        </span>
+      </div>
+      <button
```

Comment:
Could we maybe use `Button` here for consistency?

```ts
      <Button
        spotlight
        highlight
        disabled={!props.importPermitted}
      // ...
      >
        // ...
      </Button>
```

Reply:
Fixed in 19dd7446e.

Reply:
probably worth to add some custom `padding` since now it's all constrained 
<img width="59" alt="Screen Shot 2022-11-01 at 12 49 29 PM" src="https://user-images.githubusercontent.com/1081051/199226239-9718fd50-19ef-4b1a-bdc9-d38b29f91337.png">

Reply:
Fixed padding in de9b81e61.